### ISSUE TITLE
UI: Improved date selection for Task Add dialog - mockup

### ISSUE THREAD

**Issue by linuxcaffe** [CCExtractor/taskwarrior-flutter#414]
### Describe your improvement idea or feature request

When adding a new task, the current dialog offers a field for due date, however, hard due dates (when a thing MUST be done by) are not actually that common for my tasks, and I'm just as likely to want to create a new task with a sched:date, a wait:date, or even an until:date. This RFE would improve the Add dialog by allowing users to quickly and easily hit the dropdown to assign a different date type.

### How would you like to see it implemented?

add a dropdown to the date field of the Task Add dialog, leaving "due" as the default, but allowing a quick change.

### Put here any screenshots or videos (optional)

![date_selector](https://github.com/user-attachments/assets/a987cd48-c6e0-4580-919f-149208e1d146)

### Would you like to work on this issue?

None

### By submitting this issue, I have confirmed that:

- [X] I have starred the repo ⭐ and watched 👀 it on GitHub and followed the contribution guidelines.

**Comment by iitian-adi05tya** [comment]
hello @linuxcaffe can I work on this issue

**Comment by linuxcaffe** [comment]
@iitian-adi05tya sure, I don't mind, but @D-extremity might be working on something related..

**Comment by D-extremity** [comment]
Yeah, I will raise PR by Friday.
@iitian-adi05tya you can work on this, I will keep in mind about your commits and changes.

**Comment by SGI-CAPP-AT2** [comment]
Hey, 
is only one of the due, shed, wait and until attributes allowed?
because by this dropdown we can select only one of the attribute.

**Comment by linuxcaffe** [comment]
No, actually a task can have any combination of dates assigned, 
I'm working on the assumption that when a user adds a new task, that probably only one of those date types would be applicable, but right now, only the due:date is an option in that Add Dialog.
If more than one date type applies to the task, then the user can input that using the Edit Dialog.

**Comment by arjav1528** [comment]
@Pavel401 @linuxcaffe Can you assign me this issue??

**Comment by SGI-CAPP-AT2** [comment]
@arjav1528, This issue will be solved by submission of PR for issue #384 which is I am working on.

**Comment by SGI-CAPP-AT2** [comment]
@KumarADITHYA123, This is done with #442
